### ISSUE TITLE
update_by_query get no detail when conflict

### ISSUE THREAD

**Issue by okhowang** [olivere/elastic#1103]
problem is when checkResponse, it has read `res.body`
then in `newResponse` readAll will got empty string
so there is no conflict detail in response
```golang
//client.go
		// Check for errors
		if err := checkResponse((*http.Request)(req), res, opt.IgnoreErrors...); err != nil {
			// No retry if request succeeded
			// We still try to return a response.
			resp, _ = c.newResponse(res, opt.MaxResponseSize)
			return resp, err
		}
//response.go
// createResponseError creates an Error structure from the HTTP response,
// its status code and the error information sent by Elasticsearch.
func createResponseError(res *http.Response) error {
	if res.Body == nil {
		return &Error{Status: res.StatusCode}
	}
	data, err := ioutil.ReadAll(res.Body)
	if err != nil {
		return &Error{Status: res.StatusCode}
	}
	errReply := new(Error)
	err = json.Unmarshal(data, errReply)
	if err != nil {
		return &Error{Status: res.StatusCode}
	}
	if errReply != nil {
		if errReply.Status == 0 {
			errReply.Status = res.StatusCode
		}
		return errReply
	}
	return &Error{Status: res.StatusCode}
}
```
### Which version of Elastic are you using?

[ ] elastic.v7 (for Elasticsearch 7.x)
[x] elastic.v6 (for Elasticsearch 6.x)
[ ] elastic.v5 (for Elasticsearch 5.x)
[ ] elastic.v3 (for Elasticsearch 2.x)
[ ] elastic.v2 (for Elasticsearch 1.x)

### Please describe the expected behavior

get conflict detail when update by query 

### Please describe the actual behavior

get detail null

### Any steps to reproduce the behavior?

go multi update_by_query operation concurrently

**Comment by olivere** [comment]
I see. Next steps would be to come up with a test so that we can reproduce the bug and then introduce a fix.

**Comment by olivere** [comment]
I merged and backported this to v6 as well in 4a25955f5ca5a66db2e36b9a1ef9999eec06053e, 6c80d5c7e256f3687b45c0bbad2fcfb007ea1bd3 and 3ba92aba06890a842fe4a6ccf7b111dbb883c96c.
